src/dialogs/preferences/PreferenceDialog.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import Progress from '../../progress';
+import PropTypes from 'prop-types';
+import SourceItemWrapper from '../../sourceItemWrapper';
+
+/* global window */
+/* global Zotero */
+
+const PreferenceDialog = (props) => {
+    const [storage, setStorage] = useState(props.Prefs.get('storage'));
+
+    async function migrateStorageLocation(from, to) {
+        const progress = new Progress(
+            'loading',
+            props.getString('wikicite.prefs.citation-storage.progress.migrating')
+        );
+        const items = await Zotero.Items.getAll(Zotero.Libraries.userLibraryID).filter((item) => item.isRegularItem());
+        let migratedItems = 0;
+        for (let item of items) {
+            item = new SourceItemWrapper(item, from);
+            item.migrateCitations(to);
+            progress.updateLine('loading', props.getString('wikicite.prefs.citation-storage.progress.migrated-n-items', [++migratedItems, items.length]));
+        }
+        progress.updateLine('done', props.getString('wikicite.prefs.citation-storage.progress.done'));
+        progress.close();
+    }
+
+    function save() {
+        if (props.Prefs.get('storage') !== storage) {
+            // migrate citation storage location for all items
+            migrateStorageLocation(props.Prefs.get('storage'), storage).then(() => props.Prefs.set('storage', storage));
+        }
+        window.close();
+    }
+
+    function cancel() {
+        window.close();
+    }
+
+    return <div orient="vertical">
+        <fieldset>
+            <legend>{props.getString('wikicite.prefs.citation-storage')}</legend>
+            <span>{props.getString('wikicite.prefs.citation-storage-desc')}</span>
+            <br />
+            <input id="radio-storage-note" type="radio" name="citation-storage" value="note" checked={storage === 'note'} onClick={() => setStorage('note')} />
```

Comment:
I guess not a big deal, but I'm getting the following error.

```
Warning: You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`. Otherwise, set either `onChange` or `readOnly`.
```

Changing `onClick` to `onChange` fixed this error and still seems to work fine?

Reply:
Good catch! Out of interest, where did you get this error? For me it does not show up in Zotero's JS console.

Judging by the information in the warning, it seems like we should just set the initial value for the fields using ``defaultChecked``. I have changed this and it works that way. But interestingly the form fields never rendered as read-only for me anyways.

Reply:
I only saw it in the Firefox developer console, when the debug tools where attached to Zotero.

Yeah, maybe the warning doesn't account for `onClick` also being able to change the field, so it thinks it'll never be changed. In any case - that works, thanks!